.github/workflows/sui-protocol-version-monitor.yml

```diff
@@ -27,37 +27,50 @@ jobs:
+
+    - name: Download an artifact for the last saved ${{ inputs.environment }} protocol version
+      id: download-artifact
+      uses: dawidd6/action-download-artifact@v2
       with:
+        github_token: ${{secrets.GITHUB_TOKEN}}
+        workflow: sui-protocol-version-monitor.yml
+        workflow_conclusion: success
         name: ${{ inputs.environment }}-protocol-version
-    - shell: bash
+        check_artifacts: true
+        path: /tmp/${{ inputs.environment }}_saved_protocol_version.txt
+        if_no_artifact_found: warn
+
+    - name: Read last saved protocol version for ${{ inputs.environment }}
+      continue-on-error: true
+      shell: bash
       run: |
-        echo "saved_protocol_version=`cat /tmp/${{ inputs.environment }}_saved_protocol_version.txt`" >> $GITHUB_ENV
+        echo "saved_protocol_version=`cat /tmp/${{ inputs.environment }}_saved_protocol_version.txt/${{ inputs.environment }}_saved_protocol_version.txt || ''`" >> $GITHUB_ENV
 
-    - name: Read live protocol in ${{ inputs.environment }}
-      if: ${{ env.saved_protocol_version != '' }}
-      working-directory: ./
+    - name: Read live protocol version in ${{ inputs.environment }}
+      shell: bash
       run: | 
-        curl -s https://fullnode.${{ inputs.environment }}.sui.io:443 -H 'Content-Type: application/json' -d '{"jsonrpc":"2.0", "method":"suix_getLatestSuiSystemState", "params":[], "id":1}' | jq -r '.result.protocolVersion' > /tmp/${{ inputs.environment }}_current_protocol_version.txt
-        echo "current_protocol_version=`cat /tmp/${{ inputs.environment }}_current_protocol_version.txt`" >> $GITHUB_ENV
-
-    - name: Checkout sui source
-      if: ${{ env.current_protocol_version > env.saved_protocol_version }}
-      uses: actions/checkout@7dd9e2a3dc350cf687eb1b2a4fadfee8c8e49675 # pin@v3
-
+        curl -s https://fullnode.${{ inputs.environment }}.sui.io:443 -H 'Content-Type: application/json' -d '{"jsonrpc":"2.0", "method":"suix_getLatestSuiSystemState", "params":[], "id":1}' | jq -r '.result.protocolVersion' > /tmp/${{ inputs.environment }}_protocol.txt
+        echo "current_protocol_version=`cat /tmp/${{ inputs.environment }}_protocol.txt`" >> $GITHUB_ENV
 
     - name: Update framework/${{ inputs.environment }}
-      if: ${{ env.current_protocol_version > env.saved_protocol_version && github.ref ==  'refs/heads/master' }}
-      uses: Embraser01/update-git-branch-action@v1.0.0
+      if: ${{ env.saved_protocol_version != '' && env.current_protocol_version > env.saved_protocol_version }}
+      continue-on-error: true
+      uses: LucasRuy/sync-branch-action@v1.1.1
       with:
-        branch: framework//${{ inputs.environment }}
-        force: 1 # To push-force to the branch
-        githubToken: ${{ secrets.GITHUB_TOKEN }} # Github Token
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        SOURCE_BRANCH: ${{ inputs.environment }}
+        DESTINATION_BRANCH: framework/${{ inputs.environment }}
```

Comment:
I'm quite confused by the names of these variables and the [docs](https://github.com/LucasRuy/sync-branch-action) did not help, so just confirming that we're fast-forwarding `framework/<env>` to `<env>` here, rather than the other way around?

Reply:
yes, 
```
SOURCE_BRANCH | String | true | Branch where you implemented your new code.
-- | -- | -- | --
DESTINATION_BRANCH | String | true | Branch you want to send your new code.
```

Reply:
Like I said, for whatever reason the "source", "destination" description is not clicking for me -- but if it makes sense to you, and it works, all good with me!

Reply:
It does make sense to me. I'll keep an eye out to make sure it's don't what it supposed to